lib/ansible/galaxy/collection.py

```diff
@@ -1070,46 +1167,128 @@ def _build_dependency_map(collections, existing_collections, b_temp_path, apis,
     return dependency_map
 
 
+def _collections_from_scm(collection, requirement, b_temp_path, force, parent=None):
+    """
+    Returns a list of collections found in the repo. If there is a galaxy.yml in the collection then just return
+    the specific collection. Otherwise, check each top-level directory for a galaxy.yml.
+
+    :param collection: URI to a git repo
+    :param requirement: The version of the artifact
+    :param b_temp_path: The temporary path to the archive of a collection
+    :param force: Whether to overwrite an existing collection or fail
+    :param parent: The name of the parent collection
+    :return: List of CollectionRequirement objects
+    """
+
+    reqs = []
+    name, version, path, fragment = parse_scm(collection, requirement)
+    b_repo_root = to_bytes(name, errors='surrogate_or_strict')
+
+    if fragment:
+        b_fragment = to_bytes(fragment, errors='surrogate_or_strict')
+        b_collection_path = os.path.join(b_temp_path, b_repo_root, b_fragment)
+    else:
+        b_collection_path = os.path.join(b_temp_path, b_repo_root)
+
+    b_galaxy_path = os.path.join(b_collection_path, b'galaxy.yml')
```

Comment:
Not a big deal now, but I can see hard coding `galaxy.yml` in places making it slightly harder in the future to support `galaxy.yaml` when someone inevitably asks for it to be supported.

Reply:
I went ahead and added a function in 0e703ca44050eebd7468d29043dfc4c09df3f151 to get the galaxy.yml/galaxy.yaml path and updated the changelog to note that both are supported now.

Reply:
Very nice!